homeassistant/components/starlink/time.py

```diff
@@ -60,18 +61,28 @@ async def async_set_value(self, value: time) -> None:
 
 
 def _utc_minutes_to_time(utc_minutes: int, timezone: tzinfo) -> time:
-    hour = math.floor(utc_minutes / 60)
-    minute = utc_minutes % 60
-    utc = datetime.now(UTC).replace(hour=hour, minute=minute, second=0, microsecond=0)
-    return utc.astimezone(timezone).time()
+    try:
+        hour = math.floor(utc_minutes / 60)
+        minute = utc_minutes % 60
+        utc = datetime.now(UTC).replace(
+            hour=hour, minute=minute, second=0, microsecond=0
+        )
+        return utc.astimezone(timezone).time()
+    except ValueError as exc:
+        # A ValueError here means that we did not pass a valid time in to `replace` above.
```

Comment:
Is the `.replace()` call is the one throwing, that call (and only that call) should be wrapped in the try/except.

Reply:
Thanks! I wasn't sure if that was something I could do, the structure strikes me as weird